enterprise/backend/test/metabase_enterprise/serialization/v2/e2e_test.clj

```diff
@@ -735,3 +738,25 @@
                     (is (thrown-with-msg? Exception #"Please upgrade"
                                           (cmd/v2-load! dump-dir {}))
                         "throws")))))))))))
+
+(deftest every-model-is-supported-test
+  (testing "Serialization support\n"
+    (testing "We know about every model"
+      (is (= (set (concat serdes.models/exported-models
+                          serdes.models/inlined-models
+                          serdes.models/excluded-models))
+             (set (map name (v2.entity-ids/toucan-models))))))
+
+    (let [should-have-entity-id (set (concat serdes.models/data-model serdes.models/content))]
+      (doseq [model (v2.entity-ids/toucan-models)]
+        (if (contains? should-have-entity-id (name model))
+          (testing (str "Model either has entity_id or a hash key: " (name model))
+            ;; `not=` is effectively `xor`
+            (is (not= (not= (get-method serdes/entity-id (name model))
+                            (get-method serdes/entity-id :default))
+                      (and (not= (get-method serdes/hash-fields model)
+                                 (get-method serdes/hash-fields :default))
+                           (serdes.backfill/has-entity-id? model)))))
```

Comment:
I already know what condition this is supposed to express, and I can't follow this code.
```suggestion
            (let [custom-entity-id?   (not= (get-method serdes/entity-id (name model))
                                            (get-method serdes/entity-id :default))
                  custom-hash-fields? (not= (get-method serdes/hash-fields model)
                                            (get-method serdes/hash-fields :default))
                  random-entity-id?   (and custom-hash-fields?
                                           (serdes.backfill/hash-entity-id? model))]
              ;; `not=` is effectively `xor` - exactly one of these must be true.
               (is (not= custom-entity-id? random-entity-id?))
```

Reply:
done